install/nvidia

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# ==============================================================================
+# Hyprland NVIDIA Setup Script for Arch Linux
+# ==============================================================================
+# This script automates the installation and configuration of NVIDIA drivers
+# for use with Hyprland on Arch Linux, following the official Hyprland wiki.
+#
+# Author: https://github.com/Kn0ax
+# 
+# ==============================================================================
+
+# --- Colors for better readability ---
+C_RESET='\033[0m'
+C_RED='\033[0;31m'
+C_YELLOW='\033[0;33m'
+C_BOLD='\033[1m'
+
+# --- GPU Detection and Driver Selection ---
+# Get GPU info for driver selection
+gpu_info=$(lspci | grep -i 'nvidia')
+
+# Turing (16xx, 20xx), Ampere (30xx), Ada (40xx), and newer recommend the open-source kernel modules
+if echo "$gpu_info" | grep -q -E "RTX [2-9][0-9]|GTX 16"; then
+    NVIDIA_DRIVER_PACKAGE="nvidia-open-dkms"
+else
+    NVIDIA_DRIVER_PACKAGE="nvidia-dkms"
+fi
+
+# Check which kernel is installed and set appropriate headers package
+KERNEL_HEADERS="linux-headers"  # Default
+if pacman -Q linux-zen &>/dev/null; then
+    KERNEL_HEADERS="linux-zen-headers"
+elif pacman -Q linux-lts &>/dev/null; then
+    KERNEL_HEADERS="linux-lts-headers"
+elif pacman -Q linux-hardened &>/dev/null; then
+    KERNEL_HEADERS="linux-hardened-headers"
+fi
+
+# Enable multilib repository for 32-bit libraries
+if ! grep -q "^\[multilib\]" /etc/pacman.conf; then
+    sudo sed -i '/^#\[multilib\]/,/^#Include/ s/^#//' /etc/pacman.conf
+fi
+
+# Install packages
+PACKAGES_TO_INSTALL=(
+    "${KERNEL_HEADERS}"    
+    "${NVIDIA_DRIVER_PACKAGE}"
+    "nvidia-utils"
+    "lib32-nvidia-utils"
+    "egl-wayland"
+    "libva-nvidia-driver" # For VA-API hardware acceleration
+    "qt5-wayland"
+    "qt6-wayland"
+)
+
+if ! yay -Syu --needed --noconfirm "${PACKAGES_TO_INSTALL[@]}"; then
+    echo -e "${C_RED}${C_BOLD}[ERROR]${C_RESET} Failed to install NVIDIA packages. Please check the output for errors." >&2
+    exit 1
+fi
+
+# Configure modprobe for early KMS
+if ! echo "options nvidia_drm modeset=1" | sudo tee /etc/modprobe.d/nvidia.conf >/dev/null; then
+    echo -e "${C_RED}${C_BOLD}[ERROR]${C_RESET} Failed to configure modprobe. Please check your permissions." >&2
```

Comment:
This should either work or it shouldn't. We're starting from a clean slate. So think maybe this is a bit defensive.